scylla-gdb.py

```diff
@@ -5776,7 +5776,7 @@ def dump_reads_from_semaphore(semaphore):
 
         for permit in intrusive_list(permit_list):
             try:
-                schema = permit['_schema']['_p']
+                schema = schema_ptr(permit['_schema'])
```

Comment:
`schema_ptr` doesn't have a `dereference()` method which is used below.
I think you need to use `seastar_lw_shared_ptr(permit['_schema'])` instead.
Please test that it works.

Reply:
Or add dereference() to schema_ptr, it makes sense there.

Reply:
Using schema_ptr is better than explicit cast, since it abstracts the pointer type and we have one place which needs to know the implementation of schema_ptr, so coupling is reduced.

Reply:
Ok, I'm fine with adding `dereference()` to `schema_ptr()`

Reply:
`permit['_schema']['_p']` was null in some cases and I had to handle that as well. With the new fix there is no need for the dereference(). Let me know if this is okay.